File: module3.js
function calculateWork() {
  let Pext = parseFloat(document.getElementById("p-ext").value);
  let V1   = parseFloat(document.getElementById("v1").value);
  let V2   = parseFloat(document.getElementById("v2").value);

  if (isNaN(Pext) || isNaN(V1) || isNaN(V2)) {
    alert("Please enter valid numbers.");
    return;
  }

  if (V1 <= 0 || V2 <= 0) {
    alert("Volumes must be greater than zero.");
    return;
  }

  if (V1 === V2) {
    document.getElementById("output").innerHTML =
      "No expansion or compression occurred.";
    return;
  }

  // Irreversible work
  let W_irrev = -Pext * (V2 - V1);

  // Reversible work
  let n = 1;
  let R = 0.0821;
  let T = 298;
  let W_rev = -n * R * T * Math.log(V2 / V1);

  document.getElementById("output").innerHTML = `
    <p><strong>Irreversible Work:</strong> ${W_irrev.toFixed(3)} L·atm</p>
    <p><strong>Reversible Work:</strong> ${W_rev.toFixed(3)} L·atm</p>
    <br>
    <p>Initial Volume: ${V1} L</p>
    <p>Final Volume: ${V2} L</p>
    <p>External Pressure: ${Pext} atm</p>
  `;
}

File: module3/module3.py


import math

def work_calculator():
    print(" WORK CALCULATOR: REVERSIBLE vs IRREVERSIBLE EXPANSION")

    # Take inputs from user
    try:
        P_ext = float(input("Enter external pressure P_ext (in atm): "))
        V1 = float(input("Enter initial volume V1 (in L): "))
        V2 = float(input("Enter final volume V2 (in L): "))

        # For invalid inpits
        if V1 <= 0 or V2 <= 0:
            print("Volume must be greater than zero.")
            return
        if V2 == V1:
            print("No expansion or compression occurred.")
            return

    except ValueError:
        print("Please enter numbers only.")
        return

    # Irreversible Work Calculation
    W_irrev = -P_ext * (V2 - V1)

    # Reversible Work Calculation
    n = 1
    R = 0.0821 # L·atm/(mol·K)
    T = 298
    W_rev = -n * R * T * math.log(V2 / V1)

    # Output 
    print(f"""
    
                WORK CALCULATOR
    
    Initial Volume (V1): {V1} L
    Final Volume (V2): {V2} L
    External Pressure : {P_ext} atm
    
    Irreversible Work (W_irrev): {W_irrev:.3f} L·atm
    Reversible Work (W_rev) : {W_rev:.3f} L·atm
    
    """)

if __name__ == "__main__":
    work_calculator()
